enterprise/models.py

```diff
@@ -915,10 +915,22 @@ def get_paginated_content(self, query_parameters):
         Returns:
             dict: The paginated discovery service search results.
         """
+        results = []
         query = self.content_filter.copy()
         query.update(query_parameters)
         catalog_client = CourseCatalogApiServiceClient(self.enterprise_customer.site)
-        return catalog_client.get_search_results(query, traverse_pagination=False)
+        search_results = catalog_client.get_search_results(query, traverse_pagination=False)
+        for content in search_results['results']:
+            if content.get('content_type', '') == 'courserun' and content.get('has_enrollable_seats', False):
+                results.append(content)
+            elif content.get('content_type', '') == 'program' and \
+                    content.get('is_program_eligible_for_one_click_purchase', False):
+                results.append(content)
+            elif content.get('content_type', '') == 'course':
```

Comment:
Let's leave this `else` clause out for now and only include course runs and programs.